tests/builders/test_design_spaces.py

```diff
@@ -191,3 +193,99 @@ def test_exceptions(basic_cartesian_space, simple_mixture_space):
             name='invalid join space 2',
             description=''
         )
+
+
+def test_oversize_warnings(large_joint_design_space):
+    """Test that warnings are raised"""
+    # Test oversized formulation grid
+    with pytest.raises(UserWarning,
+                       match="Product design grid contains 1562500000 grid points. "
+                             "This may cause memory issues downstream."):
+        # Fail on error (so code stops running)
+        with warnings.catch_warnings():
+            warnings.simplefilter('error')
+            too_big_formulation_grid = {
+                'ing_F': np.linspace(0, 1),
```

Comment:
[Blocking] The error state depends on the default [`num=50`](https://numpy.org/doc/stable/reference/generated/numpy.linspace.html), which is a bit opaque.  Could we add that parameter explicitly in the formulations test?  You're already providing it explicitly in the enumerated space tests.

Reply:
Explicitly called out now.